File: 1922-count-good-numbers/1922-count-good-numbers.py
class Solution:
    def countGoodNumbers(self, n: int) -> int:
        mod = (10**9 + 7)
        def fn(a, n):
              if n == 0:
                return 1
              if n % 2 == 0:
                return fn(a*a % mod, n/2)
              return a * fn(a*a % mod, (n-1)/2)
        if n % 2 == 0:
            return fn(20,int(n/2)) % (10**9 + 7)
        else:
            return (5 * fn(20,n//2)) % (10**9 + 7)

File: 1922-count-good-numbers/1922-count-good-numbers.cpp

class Solution {
public:
    long long power(long long x,unsigned long long y, int p) { 
        int res = 1;   
        x = x % p; 
        if (x == 0) return 0; 

        while (y > 0) { 
            if (y & 1) res = (res*x) % p; 
            
            y = y>>1; 
            x = (x*x) % p; 
        } 
        return res; 
    } 
    int countGoodNumbers(long long n) {
        int x = 20;
        long long ans = power(20,n/2,1000000007);
        if(n % 2) ans = (ans*5) % 1000000007;
        return ans;
    }
};